pack.c

```diff
@@ -217,6 +217,7 @@ pack_pack(rb_execution_context_t *ec, VALUE ary, VALUE fmt, VALUE buffer)
     else {
         if (!RB_TYPE_P(buffer, T_STRING))
             rb_raise(rb_eTypeError, "buffer must be String, not %s", rb_obj_classname(buffer));
+        rb_str_make_independent(buffer);
```

Comment:
`rb_str_make_independent` relocates the underlying buffer, but does not check if frozen, and raises an exception when actually modifying.
This is the relocated buffer will be just wasted.

```suggestion
        rb_str_modify(buffer);
```